crates/floresta-chain/src/pruned_utreexo/consensus.rs

```diff
@@ -233,14 +378,28 @@ impl Consensus {
         };
         Ok(())
     }
-    #[allow(unused)]
+
     fn validate_locktime(
         input: &TxIn,
         transaction: &Transaction,
         height: u32,
+        mtp: u32,
     ) -> Result<(), BlockValidationErrors> {
-        unimplemented!("validate_locktime")
+        if input.sequence.is_height_locked() || input.sequence.enables_absolute_lock_time() {
+            if !transaction.is_absolute_timelock_satisfied(
+                Height::from_consensus(height).unwrap(),
+                Time::from_consensus(mtp).unwrap(),
+            ) {
+                return Err(BlockValidationErrors::BadAbsoluteLockTime);
+            }
+        }
+        if input.sequence.is_relative_lock_time() {
+            //validate lock time
+            //TODO needs to find te block that the outpoint was created
+        }
+        Ok(())
```

Comment:
NOTE: now that we have height in every utxo we can validate this easily. Implement it